apps/projects/routes.py

```diff
@@ -924,146 +992,153 @@ def post(self, project_id):
-                  example: False
-                thumbnails:
-                  type: object
-                  example: {}
-                _id:
+                  example: 640
+                height:
+                  type: integer
+                  example: 360
+                size:
+                  type: integer
+                  example: 300000
+                positoin:
                   type: string
-                  example: 5cbd5acfe24f6045607e51aa
-                preview_thumbnail:
-                  type: object
-                  properties:
-                    filename:
-                      type: string
-                      example: fa5079a38e0a4197864aa2ccb07f3bea_thumbnail.png
-                    storage_id:
-                      type: string
-                      example: 2019/5/fa5079a38e0a4197864aa2ccb07f3bea_thumbnail.png
-                    mimetype:
-                      type: string
-                      example: "image/png"
-                    width:
-                      type: integer
-                      example: 640
-                    height:
-                      type: integer
-                      example: 360
-                    size:
-                      type: int
-                      example: 300000
+                  example: custom
         """
-        doc = find_one_or_404(project_id)
-
-        check_user_agent()
-        schema = check_request_schema_validity(request.get_json(), self.SCHEMA_UPLOAD)
 
-        if doc.get('processing') is True:
-            return json_response({'processing': doc['processing']}, status=202)
+        # validate request
+        if 'file' not in request.files:
+            # to avoid TypeError: cannot serialize '_io.BufferedRandom' error
+            raise BadRequest({"file": ["required field"]})
+        document = validate_document(request.files, self.SCHEMA_UPLOAD)
 
-        base64_string = schema['data']
-        try:
-            if ',' not in base64_string:
-                base64_thumbnail = base64_string
-            else:
-                base64_format, base64_thumbnail = base64_string.split(',', 1)
-            thumbnail_stream = base64.b64decode(base64_thumbnail)
-        except base64.binascii.Error as err:
-            raise BadRequest(str(err))
-
-        video_editor = get_video_editor()
-        thumbnail_metadata = video_editor.get_meta(thumbnail_stream, 'png')
-        return self._save_thumbnail(doc, thumbnail_stream, thumbnail_metadata)
-
-    def _save_thumbnail(self, doc, stream, metadata):
-        filename, ext = os.path.splitext(doc['filename'])
-        thumbnail_filename = f"{filename}_thumbnail.png"
+        # validate codec
+        file_stream = document['file'].stream.read()
+        metadata = get_video_editor().get_meta(file_stream)
+        if metadata.get('codec_name') not in app.config.get('CODEC_SUPPORT_IMAGE'):
+            raise BadRequest(f"Codec: '{metadata.get('codec_name')}' is not supported.")
```

Comment:
this bit is duplicated, can use some helper

Reply:
We do codec validation only in 2 places, one for video and another for image, but after we reuse `file_stream` and `metadata` variables, so helper will not help here much.
I was thinking to put codec validation into custom cerberous validator, but in this case `get_video_editor().get_meta(file_stream)` will be called twice.